CMakeLists.txt

```diff
@@ -6,8 +6,9 @@ option(BUILD_PYTHON "Build the python module" ON)
 
 set(CMAKE_BUILD_TYPE RELEASE)
 set(CMAKE_CXX_STANDARD 17)
-set(CMAKE_EXE_LINKER_FLAGS "-lpthread")
 set(CMAKE_INSTALL_PREFIX ${CMAKE_SOURCE_DIR}/install)
+add_link_options(-lpthread)
+add_compile_options(-Wall -Werror)
```

Comment:
If we are going to turn on ``-Wall``, we should probably include a list of gcc versions and cray pe compiler versions in this PR that we have tested the build on to make sure users don't start getting build failures when we push to ``developer``.

Reply:
Agree, though this is independently true whether we have this PR or not